src/views/OnScreenGui.jsx

```diff
@@ -19,123 +21,110 @@ import styles from './OnScreenGui.scss';
 
-  componentWillUnmount() {
-    window.removeEventListener('keydown', this.toggleConsole);
-  }
+  React.useEffect(() => {
+    startConnection();
+    window.addEventListener("keydown", toggleConsole);
+    return () => window.removeEventListener('keydown', toggleConsole);
+  }, []);
 
-  toggleConsole(e) {
+ 
+  function toggleConsole(e) {
       if(e.code === "Backquote") {
-        this.setState({
-          luaConsoleVisible : !this.state.luaConsoleVisible
-        });
+        setLuaConsoleVisible(!luaConsoleVisible);
       }
   }
 
-  checkVersion() {
-    if (!this.checkedVersion && this.props.version.isInitialized) {
-      const versionData = this.props.version.data;
-      if (!isCompatible(versionData.openSpaceVersion, RequiredOpenSpaceVersion)) {
-        console.warn(
-          'Possible incompatibility: \nRequired OpenSpace version: ' +
-          formatVersion(RequiredOpenSpaceVersion) +
-          '. Currently controlling OpenSpace version ' +
-          formatVersion(versionData.openSpaceVersion) + '.'
-        );
-      }
-      if (!isCompatible(versionData.socketApiVersion, RequiredSocketApiVersion)) {
-        console.warn(
-          "Possible incompatibility: \nRequired Socket API version: " +
-          formatVersion(RequiredSocketApiVersion) +
-          ". Currently operating over API version " +
-          formatVersion(versionData.socketApiVersion) + '.'
-        );
-      }
-      this.checkedVersion = true;
+  // Check the version
+  if (!hasCheckedVersion && version.isInitialized) {
+    const versionData = version.data;
+    if (!isCompatible(versionData.openSpaceVersion, RequiredOpenSpaceVersion)) {
+      console.warn(
+        `Possible incompatibility: \nRequired OpenSpace version: ${
+          formatVersion(RequiredOpenSpaceVersion)
+        }. Currently controlling OpenSpace version ${
+          formatVersion(versionData.openSpaceVersion)}.`,
+      );
+    }
+    if (!isCompatible(versionData.socketApiVersion, RequiredSocketApiVersion)) {
+      console.warn(
+        `Possible incompatibility: \nRequired Socket API version: ${
+          formatVersion(RequiredSocketApiVersion)
+        }. Currently operating over API version ${
+          formatVersion(versionData.socketApiVersion)}.`,
+      );
     }
+    hasCheckedVersion = true;
   }
 
-  reloadGui() {
+  function reloadGui() {
     location.reload();
   }
 
-  render() {
-    const {isInBrowser} = this.props;
-    this.checkVersion();
-    return (
-      <div className={styles.app}>
-        <TutorialProvider>
-          {this.props.showAbout && (
+  return (
+    <div className={styles.app}>
+      <TutorialProvider>
+        { showAbout && (
           <Overlay>
             <Stack style={{ maxWidth: '500px' }}>
-              <Button style={{ alignSelf: 'flex-end', color: 'white' }} onClick={this.props.hideAbout}>
+              <Button style={{ alignSelf: 'flex-end', color: 'white' }} onClick={hideAbout}>
```

Comment:
line width 

I'd argue that even in the GUI code we should try to aim for a maximum line width of 90 characters

I know that there are already a lot of places in the code where the lines are already too long... But why not leave the code better than we found it when doing changes :)